e2e-tests/contentful/snapshots.js

```diff
@@ -1,58 +1,58 @@
 module.exports = {
   "content-reference": {
     "content-reference-many-2nd-level-loop": {
-      "1": "<div data-cy-id=\"content-reference-many-2nd-level-loop\">\n  <h2>Content Reference: Many (2nd level loop)</h2>\n  <p data-cy-id=\"integer\">[ContentfulInteger] 42</p>\n  <p data-cy-id=\"text\">[ContentfulText] The quick brown fox jumps over the lazy\n    dog.</p>\n  <p data-cy-id=\"reference\">[ContentfulReference] Content Reference: One (Loop A\n    -&gt; B): [Content Reference: One (Loop B -&gt; A)]</p>\n</div>"
+      "1": "<div data-cy-id=\"content-reference-many-2nd-level-loop\">\n  <h2>Content Reference: Many (2nd level loop)</h2>\n  <p data-cy-id=\"integer\">[ContentfulInteger]\n    <!-- -->42</p>\n  <p data-cy-id=\"text\">[ContentfulText]\n    <!-- -->The quick brown fox jumps over the lazy dog.</p>\n  <p data-cy-id=\"reference\">[ContentfulReference]\n    <!-- -->Content Reference: One (Loop A -&gt; B)\n    <!-- -->: [\n    <!-- -->Content Reference: One (Loop B -&gt; A)\n    <!-- -->]</p>\n</div>"
```

Comment:
These empty comments (`<!-- -->`) appear when I render regular html/jsx or rich text.

Reply:
Related code for rendering:

https://github.com/gatsbyjs/gatsby/blob/master/e2e-tests/contentful/src/components/references/content-reference.js#L15-L18

Reply:
I think comment nodes are created by React to demarcate different React elements that produce a single text node in DOM. See https://github.com/facebook/react/issues/14725

But the question is why do we see them now.

Reply:
Did we change to `renderToString` from something else lately? I was told something to the rendering was changed not to long ago.

Pretty sure the initial tests have been created with Gatsby v3 already.

Reply:
But that would basically mean updating these snapshots is fine?

Reply:
The tests did not fail in the commit that might introduces this change because of https://github.com/gatsbyjs/gatsby/pull/31073#pullrequestreview-645890567

Reply:
My best guess is that cypress now snapshots HTML before React is hydrated. Maybe `"runMode": 2` caused final snapshots to always occur after hydration and that's why we see this difference.

But if it is the case - then it is potentially flaky.

Reply:
I guess we can try to confirm this hypothesis by adding some timeout before doing a snapshot?

Reply:
I've poked around with this and looks like React preserves those empty HTML comments (originally created with SSR) even with subsequent client-side renders.

So in short, this seems to be a valid snapshot and shouldn't be flaky.

I still don't get why the previous version has not had those empty comments. In theory, the only way this could have happened is if this page transition in cypress was client-side:

https://github.com/gatsbyjs/gatsby/blob/ae2e2de0e4da03d7a0a662a9cf3af15d26c1e741/e2e-tests/contentful/cypress/integration/content-reference.js#L3

Reply:
There might have been a reformatting of the source file which leads to this. I used [] which prettier seems to prefer to move into a new line.

I've never seen the non-comment variant again. I don't think its flaky